egs/wsj/s5/steps/nnet3/train_raw_dnn.py

```diff
@@ -0,0 +1,330 @@
+
+        run_opts.train_queue_opt = "--gpu 1"
+        run_opts.parallel_train_opts = ""
+        run_opts.combine_queue_opt = "--gpu 1"
+        run_opts.prior_gpu_opt = "--use-gpu=yes"
+        run_opts.prior_queue_opt = "--gpu 1"
+
+    else:
+        logger.warning("""
+    Without using a GPU this will be very slow.  nnet3 does not yet support multiple threads.""")
+
+        run_opts.train_queue_opt = ""
+        run_opts.parallel_train_opts = "--use-gpu=no"
+        run_opts.combine_queue_opt = ""
+        run_opts.prior_gpu_opt = "--use-gpu=no"
+        run_opts.prior_queue_opt = ""
+
+    run_opts.command = args.command
+    run_opts.egs_command = args.egs_command if args.egs_command is not None else args.command
+    run_opts.num_jobs_compute_prior = args.num_jobs_compute_prior
+
+    return [args, run_opts]
+
+# args is a Namespace with the required parameters
+def Train(args, run_opts):
+    arg_string = pprint.pformat(vars(args))
+    logger.info("Arguments for the experiment\n{0}".format(arg_string))
+
+    # Set some variables.
+    feat_dim = GetFeatDim(args.feat_dir)
+    ivector_dim = GetIvectorDim(args.online_ivector_dir)
+
+    # split the training data into parts for individual jobs
+    SplitData(args.feat_dir, args.nj)
+
+    config_dir = '{0}/configs'.format(args.dir)
+    var_file = '{0}/vars'.format(config_dir)
+
+    variables = ParseModelConfigGenericVarsFile(var_file)
+
+    # Set some variables.
+
+    try:
+        left_context = variables['model_left_context']
+        right_context = variables['model_right_context']
+        num_hidden_layers = variables['num_hidden_layers']
+        num_targets = int(variables['num_targets'])
+        add_lda = StrToBool(variables['add_lda'])
+        include_log_softmax = StrToBool(variables['include_log_softmax'])
+        objective_type = variables['objective_type']
+    except KeyError as e:
+        raise Exception("KeyError {0}: Variables need to be defined in {1}".format(
+            str(e), '{0}/configs'.format(args.dir)))
+    # Initialize as "raw" nnet, prior to training the LDA-like preconditioning
+    # matrix.  This first config just does any initial splicing that we do;
+    # we do this as it's a convenient way to get the stats for the 'lda-like'
+    # transform.
+
+    if args.use_dense_targets:
+        if GetFeatDimFromScp(targets_scp) != num_targets:
+            raise Exception("Mismatch between num-targets provided to "
+                            "script vs configs")
+
+    if (args.stage <= -5):
+        logger.info("Initializing a basic network for estimating preconditioning matrix")
+        RunKaldiCommand("""
+{command} {dir}/log/nnet_init.log \
+    nnet3-init --srand=-2 {dir}/configs/init.config {dir}/init.raw
+    """.format(command = run_opts.command,
+               dir = args.dir))
+
+    default_egs_dir = '{0}/egs'.format(args.dir)
+
+    if args.use_dense_targets:
+        target_type = "dense"
+        compute_accuracy = False
```

Comment:
add a comment here describing why compute_accuracy is false for dense type. This is not readily obvious as even dense posterior targets can exist.